tests/components/samsungtv/test_device_trigger.py

```diff
@@ -0,0 +1,153 @@
+
+from tests.common import MockConfigEntry, async_get_device_automations
+
+
+@pytest.mark.usefixtures("remoteencws", "rest_api")
+async def test_get_triggers(hass: HomeAssistant) -> None:
+    """Test we get the expected triggers."""
+    await setup_samsungtv_entry(hass, MOCK_ENTRYDATA_ENCRYPTED_WS)
+
+    device_reg = get_dev_reg(hass)
+    device = device_reg.async_get_device(identifiers={(DOMAIN, "any")})
+
+    turn_on_trigger = {
+        "platform": "device",
+        "domain": DOMAIN,
+        "type": "samsungtv.turn_on",
+        "device_id": device.id,
+        "metadata": {},
+    }
+
+    triggers = await async_get_device_automations(
+        hass, DeviceAutomationType.TRIGGER, device.id
+    )
+    assert turn_on_trigger in triggers
+
+
+@pytest.mark.usefixtures("remoteencws", "rest_api")
+async def test_if_fires_on_turn_on_request(
+    hass: HomeAssistant, calls: list[ServiceCall]
+) -> None:
+    """Test for turn_on and turn_off triggers firing."""
+    await setup_samsungtv_entry(hass, MOCK_ENTRYDATA_ENCRYPTED_WS)
+
+    device_reg = get_dev_reg(hass)
+    device = device_reg.async_get_device(identifiers={(DOMAIN, "any")})
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        "platform": "device",
+                        "domain": DOMAIN,
+                        "device_id": device.id,
+                        "type": "samsungtv.turn_on",
+                    },
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {
+                            "some": "{{ trigger.device_id }}",
+                            "id": "{{ trigger.id }}",
+                        },
+                    },
+                },
+                {
+                    "trigger": {
+                        "platform": "samsungtv.turn_on",
+                        "entity_id": ENTITY_ID,
+                    },
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {
+                            "some": ENTITY_ID,
+                            "id": "{{ trigger.id }}",
+                        },
+                    },
+                },
+            ],
+        },
+    )
+
+    with patch(
+        "homeassistant.components.samsungtv.media_player.send_magic_packet"
+    ) as mock_send_magic_packet:
+        await hass.services.async_call(
+            "media_player",
+            "turn_on",
+            {"entity_id": ENTITY_ID},
+            blocking=True,
+        )
+
+    await hass.async_block_till_done()
+    assert len(calls) == 2
+    assert calls[0].data["some"] == device.id
+    assert calls[0].data["id"] == 0
+    assert calls[1].data["some"] == ENTITY_ID
+    assert calls[1].data["id"] == 0
+
+    assert mock_send_magic_packet.called
```

Comment:
I suggest we don't assert the magic packet call as it's not important for the trigger feature and the magic packet feature should also be deprecated probably, in the future, as it doesn't belong in another device integration.